lib/elixir/lib/map_set.ex

```diff
@@ -18,8 +18,44 @@ defmodule MapSet do
   @opaque t :: %__MODULE__{map: map}
   defstruct map: %{}
 
+  @spec new :: %MapSet{}
   def new(), do: %MapSet{}
 
+  @doc """
+  Creates a `MapSet` from an enumerable.
+
+  Duplicated entries are removed.
+
+  ## Examples
+
+      iex> MapSet.new([:b, :a, 3])
+      #MapSet<[3, :a, :b]>
+
+      iex> MapSet.new([3, 3, 3, 2, 2, 1])
+      #MapSet<[1, 2, 3]>
+
+  """
+  @spec new(Enum.t) :: %MapSet{}
+  def new(collection) do
```

Comment:
`enumerable` and `collectable` are variable names used in Enum, so we should follow that convention